### PR TITLE
enhance(runtime): starwhale runtime standardization(runtime build = lock + package)

### PR DESCRIPTION
**Pull request by tianweidut**
## Description
- related: https://github.com/star-whale/starwhale/issues/1418
- **Starwhale Runtime Build = lock + package**
  - three environment selectors:
     - `env_name`: name, only support conda
     - `env_prefix_path`: conda or venv prefix path, if omitted it, `PWD/.starwhale/{conda|venv}` will be created automatically. 
     - `env_use_shell`: use current shell
  - main build process:
     1.  [optional] create an isolated python environment in the  `PWD/.starwhale/{conda|venv}`
     2.  install runtime.yaml dependencies: conda pkgs, conda env yaml, pip pkgs, pip requirements.txt
     3.  lock environment file:  `pip freeze > requirements-sw-lock.txt`  or `conda export > conda-sw-lock.yaml`
     4.  build a swrt package

### 1.quickstart cmd
- quickstart from shell for venv
  [asciicast](https://asciinema.org/a/534797)
- quickstart from existed runtime uri for  venv
  [asciicast](https://asciinema.org/a/534824)
- quickstart from shell for conda
  [asciicast](https://asciinema.org/a/534839)
### 2.runtime build command
- build after quickstart
  [asciicast](https://asciinema.org/a/534803)
- build from the only runtime.yaml
  [asciicast](https://asciinema.org/a/534843)

### 3. lock command
- venv lock
  [asciicast](https://asciinema.org/a/534815)
- conda lock
  [asciicast](https://asciinema.org/a/534847)

## Modules
- [x] Client

## Checklist
- [x] run code format and lint check
- [x] add unit test
- [ ] add necessary doc

### CODE REVIEW

File: client/starwhale/core/runtime/model.py [hunk 1]
```diff
@@ -1475,7 +1545,33 @@ def restore(
...
+            f":baby_bottle: install runtime.yaml dependencies @ {env_dir} for lock..."
+        )
+        deps_config = Dependencies(runtime_yaml.get("dependencies", []))
+        for dep in deps_config.deps:
+            if dep.kind in skip_deps:
+                logger.debug(f"skip {dep} to install")
+                continue
+
+            _func = (
+                dep.conda_install if PythonRunEnv.CONDA == mode else dep.venv_install
```
Comment (jialeicui) [inline client/starwhale/core/runtime/model.py:1569]: What will happen when "use current system shell"?
Will `venv_install` make unexpected behavior?

File: client/starwhale/core/runtime/model.py [hunk 2]
```diff
@@ -1176,58 +1223,73 @@ def lock(
...
+            _sw_auto_path = target_dir / SW_AUTO_DIRNAME / mode
+            cls._ensure_isolated_python_env(_sw_auto_path, expected_pyver, mode)
+            prefix_path = str(_sw_auto_path)
+
+        # TODO: support wheel pre-install for dump wheel dependencies?
+        cls._install_dependencies_with_runtime_yaml(
+            workdir=target_dir,
+            runtime_yaml=runtime_yaml,
+            env_dir=prefix_path,
+            skip_deps=[DependencyType.WHEEL, DependencyType.NATIVE_FILE],
```
[the remark this answers is not in the document]
Comment (tianweidut) [reply client/starwhale/core/runtime/model.py:1265]: ok

[the remark this answers is not in the document]
Comment (tianweidut) [reply client/starwhale/core/runtime/model.py]: done

File: client/starwhale/core/runtime/model.py (hunk 1 above)
Comment (tianweidut) [reply client/starwhale/core/runtime/model.py:1568]: when users config `env_use_shell=True`:
1. get prefix path by [env](https://github.com/star-whale/starwhale/pull/1450/files#diff-368402d5a9ece27e6cf34e94a5028e75d935c7fc0712dd10f4868800c749c780R1252) 
2. use prefix path bin to install pkgs.

**Review by jialeicui — APPROVED** [review]
LGTM

**Pull request merged**